src/main/java/com/minecolonies/entity/ai/citizen/guard/EntityAIRangeGuard.java

```diff
@@ -0,0 +1,300 @@
+     */
+    private static final double MAX_ATTACK_DISTANCE = 20.0D;
+
+    /**
+     * Damage per range attack.
+     */
+    private static final int DAMAGE_PER_ATTACK = 2;
+
+    /**
+     * When target is out of sight, try to move that close to the target.
+     */
+    private static final int MOVE_CLOSE = 3;
+
+    /**
+     * Sets up some important skeleton stuff for every ai.
+     *
+     * @param job the job class
+     */
+    public EntityAIRangeGuard(@NotNull final JobGuard job)
+    {
+        super(job);
+        super.registerTargets(
+          new AITarget(GUARD_SEARCH_TARGET, this::searchTarget),
+          new AITarget(GUARD_GET_TARGET, this::getTarget),
+          new AITarget(GUARD_HUNT_DOWN_TARGET, this::huntDown),
+          new AITarget(GUARD_PATROL, this::patrol),
+          new AITarget(GUARD_RESTOCK, this::goToBuilding)
+        );
+
+        if (worker.getCitizenData() != null)
+        {
+            worker.setSkillModifier(2 * worker.getCitizenData().getIntelligence() + worker.getCitizenData().getStrength());
+            worker.setCanPickUpLoot(true);
+        }
+    }
+
+    @Override
+    protected AIState searchTarget()
+    {
+        if (checkOrRequestItems(new ItemStack(Items.BOW)))
+        {
+            return AIState.GUARD_SEARCH_TARGET;
+        }
+        worker.setHeldItem(worker.findFirstSlotInInventoryWith(Items.BOW));
+        return super.searchTarget();
+    }
+
+    /**
+     * Follow the target and kill it.
+     *
+     * @return the next AIState.
+     */
+    protected AIState huntDown()
+    {
+        if (!targetEntity.isEntityAlive() || checkOrRequestItems(new ItemStack(Items.BOW)))
+        {
+            targetEntity = null;
+        }
+
+        if (targetEntity != null)
+        {
+            if (worker.getEntitySenses().canSee(targetEntity) && worker.getDistanceToEntity(targetEntity) <= MAX_ATTACK_DISTANCE)
+            {
+                worker.resetActiveHand();
+                attackEntityWithRangedAttack(targetEntity, DAMAGE_PER_ATTACK);
+                setDelay(getReloadTime());
+                attacksExecuted += 1;
+
+                if (attacksExecuted >= getMaxAttacksUntilRestock())
+                {
+                    return AIState.GUARD_RESTOCK;
+                }
+
+                return AIState.GUARD_HUNT_DOWN_TARGET;
+            }
+            worker.setAIMoveSpeed((float) (BASE_FOLLOW_SPEED + BASE_FOLLOW_SPEED_MULTIPLIER * worker.getExperienceLevel()));
+            worker.isWorkerAtSiteWithMove(targetEntity.getPosition(), MOVE_CLOSE);
+
+            return AIState.GUARD_SEARCH_TARGET;
+        }
+
+        worker.setAIMoveSpeed(1);
+        return AIState.GUARD_SEARCH_TARGET;
+    }
+
+    @Override
+    public void attackEntityWithRangedAttack(@NotNull EntityLivingBase entityToAttack, float baseDamage)
+    {
+        EntityTippedArrow arrowEntity = new EntityTippedArrow(this.worker.worldObj, worker);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'arrowEntity' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)